File: Array/121-Best-Time-to-Buy-and-Sell-Stock.ts
function maxProfit(prices: number[]): number {
    let maxValue = 0;
    let l = 0;
    let r = l + 1;

    while (r < prices.length) {
        const value = prices[r] - prices[l];
        if (value > 0) {
            maxValue = Math.max(maxValue, value);
        } else {
            l = r;
        }

        r++;
    }

    return maxValue;
};

File: Array/121-Best-Time-to-Buy-and-Sell-Stock/121-Best-Time-to-Buy-and-Sell-Stock.py
def maxProfit(prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        res = 0
        l = 0
        r = l + 1
        
        while (r < len(prices)):
            profit = prices[r] - prices[l]
            if profit > 0:
                res = max(res, profit)
            else:
                l = r
        
            r += 1
            
        return res
        
        